File: 02_Arrays and Hashing/02_ContainsDuplicate/02_ContainsDuplicate.cpp
#include<iostream>
#include<unordered_map>
#include<vector>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> umap;

        int i = 0;
        while (i < nums.size()) {
            umap[nums[i]]++;
            i++;
        }
        
        i = 0;
        while (i < nums.size()) {
            if (umap[nums[i]] > 1) return true;
            i++;
        }
        
        return false;
        
    }
};


int main() {
    Solution s;
    vector<int> arg = {1, 2, 1, 3, 1, 4};
    cout << "Contains Duplicate: " << s.containsDuplicate(arg) << endl;
    return 0;
}

File: 02_Arrays and Hashing/02_ContainsDuplicate/02_ContainsDuplicate.js
// https://leetcode.com/problems/contains-duplicate/description/

function containsDuplicate(nums) {
    const mp = {};
    let i = 0;
    while (i < nums.length) {
        if (mp[nums[i]] == undefined) {
            mp[nums[i]] = 1;
        } else {
            mp[nums[i]]++;
        }
        i++;
    }
    i = 0;
    while (i < nums.length) {
        if (mp[nums[i]] > 1) {
            return true;
        }
        i++;
    }
    return false;
}


console.log(containsDuplicate([1,1,1,3,3,4,3,2,4,2]));